pandas/core/arrays/arrow/array.py

```diff
@@ -2368,20 +2368,27 @@ def _str_fullmatch(
         return self._str_match(pat, case, flags, na)
 
     def _str_find(self, sub: str, start: int = 0, end: int | None = None) -> Self:
-        if start != 0 and end is not None:
+        if (start == 0 or start is None) and end is None:
```

Comment:
Any chance you've looked at using a slice object here already? It feels like that could help simplify a lot of the branching being done here for things being None / 0

Reply:
I don't think a slice object would help here, since I believe for this method, we need a positive start index, to calculate the offset of the substring from the original string. This PR essentially converts start into equivalent, but positive index, then it can be added to the resulting index returned by pc.find_substring. For example, if string is "abc" and start is -1, we calculate start_offset as 2, which is the equivalent positive index. We can use this offset to find the index of the substring in the original string, by adding it the position of the substring result.